### PR TITLE
feat: floating local video view is draggable

### PR DESCRIPTION
**Pull request by vanGalilea**
The goal is to have the local video floating view draggable to view partic. when needed.

https://github.com/GetStream/stream-video-js/assets/25864161/ab8cb297-4dd9-4301-bd41-6100f6debfe8

Limitations: no snapping to top right/left, bottom right/left corners

### CODE REVIEW

File: packages/react-native-sdk/src/components/LocalVideoView.tsx [hunk 1]
```diff
@@ -56,29 +63,59 @@ export const LocalVideoView = (props: LocalVideoViewProps) => {
   const isCameraOnFrontFacingMode = useStreamVideoStoreValue(
     (store) => store.isCameraOnFrontFacingMode,
   );
+  const pan = useRef(new Animated.ValueXY()).current;
+  const panResponder = useRef(
+    PanResponder.create({
+      onMoveShouldSetPanResponder: () => true,
+      onPanResponderMove: Animated.event([null, { dx: pan.x, dy: pan.y }], {
+        useNativeDriver: false,
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/LocalVideoView.tsx:71]: why is it false here? is true not supported for this usecase?

false is potentially slow on medium/low end android devices

File: packages/react-native-sdk/src/components/LocalVideoView.tsx [hunk 2]
```diff
@@ -56,29 +63,59 @@ export const LocalVideoView = (props: LocalVideoViewProps) => {
   const isCameraOnFrontFacingMode = useStreamVideoStoreValue(
     (store) => store.isCameraOnFrontFacingMode,
   );
+  const pan = useRef(new Animated.ValueXY()).current;
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/LocalVideoView.tsx:66]: some comments of what this variable does and the animation logic would help for maintenance

File: packages/react-native-sdk/src/components/LocalVideoView.tsx [hunk 3]
```diff
@@ -56,29 +63,59 @@ export const LocalVideoView = (props: LocalVideoViewProps) => {
...
+  const panResponder = useRef(
+    PanResponder.create({
+      onMoveShouldSetPanResponder: () => true,
+      onPanResponderMove: Animated.event([null, { dx: pan.x, dy: pan.y }], {
+        useNativeDriver: false,
+      }),
+      onPanResponderRelease: () => {
+        pan.extractOffset();
```
Comment (santhoshvai) [inline packages/react-native-sdk/src/components/LocalVideoView.tsx:74]: what does this do? 

btw with the release, we can snap to positions... Can this be added in this PR?

File: packages/react-native-sdk/src/components/LocalVideoView.tsx (hunk 1 above)
Comment (santhoshvai) [reply packages/react-native-sdk/src/components/LocalVideoView.tsx:71]: translate animations are supported with useNativeDrive afaik

File: packages/react-native-sdk/src/components/LocalVideoView.tsx (hunk 3 above)
Comment (vanGalilea) [reply packages/react-native-sdk/src/components/LocalVideoView.tsx:74]: > Limitations: no snapping to top right/left, bottom right/left corners.

Agree, and I mentioned the limitation indeed, going to timebox the snapping feat.

File: packages/react-native-sdk/src/components/LocalVideoView.tsx (hunk 1 above)
Comment (vanGalilea) [reply packages/react-native-sdk/src/components/LocalVideoView.tsx:71]: they are supported, it just failed due to an unknow reason now it works 👍🏻

Comment (vanGalilea) [reply packages/react-native-sdk/src/components/LocalVideoView.tsx:71]: https://github.com/facebook/react-native/issues/13377#issuecomment-292657541

File: packages/react-native-sdk/src/components/LocalVideoView.tsx (hunk 3 above)
Comment (vanGalilea) [reply packages/react-native-sdk/src/components/LocalVideoView.tsx:74]: This will take to much time to figure out atm, going to park this ;)

**Pull request merged**